### PR TITLE
[MRG] renamed distributions for consistency with R and statsmodels

### PR DESCRIPTION
**Pull request by pavanramkumar**

### CODE REVIEW

**Comment by pavanramkumar** [comment]
closes #125

File: tests/test_pyglmnet.py [hunk 1]
```diff
@@ -44,28 +44,38 @@ def test_tikhonov():
...
-    R2_train['tikhonov'] = glm_tikhonov[-1].score(ytrain, ytrain_hat, np.mean(ytrain), method='pseudo_R2')
-    R2_test['tikhonov'] = glm_tikhonov[-1].score(ytest, ytest_hat, np.mean(ytrain), method='pseudo_R2')
+    R2_train['tikhonov'] = glm_tikhonov[-1].score(Xtrain, ytrain)
+    R2_test['tikhonov'] = glm_tikhonov[-1].score(Xtest, ytest)
 
     # fit model with cdfast
-    glm_tikhonov = GLM(distr='poisson', alpha=0.0, Tau=Tau, solver='cdfast', tol=1e-5)
+    glm_tikhonov = GLM(distr='softplus',
+                       alpha=0.0,
+                       Tau=Tau,
```
Comment (jasmainak) [inline tests/test_pyglmnet.py]: woops, uppercase. @pavanramkumar did we remove pyflakes checks?

File: pyglmnet/pyglmnet.py [hunk 2]
```diff
@@ -678,47 +688,100 @@ def fit_predict(self, X, y):
...
-        method : str
-            One of 'pseudo_R2' or 'deviance'
+
+        Returns
+        -------
+        score: array | float
+            array when score is called by a list of estimators `glm.score()`
+            float when score is called by a sliced estimator `glm[0].score()`
+
+            Note that if you want compatibility with sciki-learn's
```
Comment (jasmainak) [inline pyglmnet/pyglmnet.py]: maybe use `note` from sphinx: http://www.sphinx-doc.org/en/stable/markup/para.html ?

File: pyglmnet/pyglmnet.py [hunk 3]
```diff
@@ -678,47 +688,100 @@ def fit_predict(self, X, y):
...
+            pipeline, cross_val_score, or GridSearchCV then you should
+            only pass sliced estimators:
+
+            from sklearn.grid_search import GridSearchCV
```
Comment (jasmainak) [inline pyglmnet/pyglmnet.py]: does this render as python code?

File: examples/api/plot_softplus.py [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+exponential nonlinearity. However, the gradient of the loss function that uses
+this nonlinearity is typically unstable.
+
+This is an example demonstrating how pyglmnet
+works with a Poisson distribution using the softplus nonlinearity.
+
+Here, for the ``distr = 'softplus'`` option, we use the
+softplus function: ``log(1+exp())``.
+
+"""
```
Comment (jasmainak) [inline examples/api/plot_softplus.py]: actually, it just occured to me. Can we call the folder the example is in something other than API? It's a bit confusing to me ...

File: examples/api/plot_softplus.py [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+
+##########################################################
+
+fit_param = glm_poisson[-1].fit_
+plt.plot(beta[:], 'bo', label ='true')
+plt.plot(fit_param['beta'][:], 'ro', label='estimated')
+plt.xlabel('samples')
+plt.ylabel('outputs')
+plt.legend(bbox_to_anchor=(0., 1.02, 1., .102), loc=1,
+           ncol=2, borderaxespad=0.)
```
Comment (jasmainak) [inline examples/api/plot_softplus.py]: we don't need publication quality figures in examples. The code should be simple so as to not scare new users :)

File: examples/api/plot_softplus.py [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+# Predict targets from test set
+yrhat = glm_poisson[-1].predict(scaler.transform(Xr))
+ythat = glm_poisson[-1].predict(scaler.transform(Xt))
+
+plt.plot(yt[:100], label='true')
+plt.plot(ythat[:100], 'r', label='predicted')
+plt.xlabel('samples')
+plt.ylabel('true and predicted outputs')
+plt.legend(bbox_to_anchor=(0., 1.02, 1., .102), loc=1,
+           ncol=2, borderaxespad=0.)
```
Comment (jasmainak) [inline examples/api/plot_softplus.py]: same comment here

File: pyglmnet/pyglmnet.py [hunk 7]
```diff
@@ -147,8 +150,10 @@ def __init__(self, distr='poisson', alpha=0.5,
         self.learning_rate = learning_rate
         self.max_iter = max_iter
         self.fit_ = None
+        self.ynull_ = None
         self.tol = tol
         self.eta = eta
+        self.score_metric = score_metric
```
Comment (jasmainak) [inline pyglmnet/pyglmnet.py]: you need a check to throw an error if it's not `deviance` or `pseudo_R2`

File: pyglmnet/pyglmnet.py [hunk 8]
```diff
@@ -678,47 +688,100 @@ def fit_predict(self, X, y):
...
+
+            if self.distr in ['softplus', 'poisson']:
+                LS = utils.log_likelihood(y, y, self.distr)
             else:
-                score = 1 - L1 / L0
+                LS = 0
+            if(self.score_metric == 'pseudo_R2'):
+                L0 = utils.log_likelihood(y, self.ynull_, self.distr)
+
+            for idx in range(yhat.shape[0]):
```
Comment (jasmainak) [inline pyglmnet/pyglmnet.py]: any chance of making the code more DRY? I see that you're doing the same thing in the two conditions. If you do:

```py
if len(yhat.shape) == 1:
    yhat = yhat[np.newaxis, ...]
```
and then L727-747 are basically taken care of by L763-775, no?

File: examples/api/plot_poisson.py [hunk 9]
```diff
@@ -26,143 +24,156 @@
...
+#         \exp(z_i), & \text{if}\ z_i \leq \eta \\
+#         \\
+#          \exp(\eta)z_i + (1-\eta)\exp(\eta), & \text{if}\ z_i \gt \eta
+#     \end{cases}
+#
+# where
+#
+# .. math::
+#
+#     z_i = \beta_0 + \sum_j \beta_j x_{ij}
```
Comment (jasmainak) [inline examples/api/plot_poisson.py]: shouldn't the plot labels be changed to match this equation?

File: examples/api/plot_poisson.py [hunk 10]
```diff
@@ -26,143 +24,156 @@
...
+#
+#     J = \sum_i \lambda_i - y_i \log \lambda_i
+#
+# Instead of the canonical link, we define \lambda as follows:
+#
+# .. math::
+#
+#     \lambda_i =
+#     \begin{cases}
+#         \exp(z_i), & \text{if}\ z_i \leq \eta \\
```
Comment (jasmainak) [inline examples/api/plot_poisson.py]: you have not defined `\eta` anywhere although you have said it in words

File: examples/api/plot_poisson.py [hunk 11]
```diff
@@ -26,143 +24,156 @@
...
+#         \exp(\eta) \sum_i \Big(1 - \frac{\lambda_i}{y_i}\Big), & \text{if}\ z_i \gt \eta
+#     \end{cases}
+#
+# .. math::
+#     \frac{\partial J}{\partial \beta_j} =
+#     \begin{cases}
+#         \sum_i \Big(\lambda_i - y_i\Big)x_{ij}, & \text{if}\ z_i \leq \eta \\
+#         \\
+#         \exp(\eta) \sum_i \Big(1 - \frac{\lambda_i}{y_i}\Big)x_{ij}, & \text{if}\ z_i \gt \eta
+#     \end{cases}
```
Comment (jasmainak) [inline examples/api/plot_poisson.py]: It's not super clear to me why you derive the derivatives in this example ...

**Comment by jasmainak** [comment]
That's it for now @pavanramkumar ! We're getting there ... :)

File: tests/test_pyglmnet.py (hunk 1 above)
Comment (pavanramkumar) [inline tests/test_pyglmnet.py]: This was part of PR #150 and `Tau` is a matrix, so I chose to use upper case. We have only `flake8` in CI.

File: pyglmnet/pyglmnet.py (hunk 7 above)
Comment (pavanramkumar) [inline pyglmnet/pyglmnet.py]: Thanks, done!

File: pyglmnet/pyglmnet.py (hunk 2 above)
Comment (pavanramkumar) [inline pyglmnet/pyglmnet.py]: Can docstring have rich formatting?

File: pyglmnet/pyglmnet.py (hunk 3 above)
Comment (pavanramkumar) [inline pyglmnet/pyglmnet.py]: No, it's in the docstring. Can docstring have rich formatting?

File: pyglmnet/pyglmnet.py (hunk 8 above)
Comment (pavanramkumar) [inline pyglmnet/pyglmnet.py]: Great point, done now.

File: examples/api/plot_softplus.py (hunk 4 above)
Comment (pavanramkumar) [inline examples/api/plot_softplus.py]: Sure, but in another PR where I'll make examples more readable and neat

File: pyglmnet/pyglmnet.py (hunk 2 above)
Comment (jasmainak) [inline pyglmnet/pyglmnet.py]: yes, it can!

File: pyglmnet/pyglmnet.py (hunk 3 above)
Comment (jasmainak) [inline pyglmnet/pyglmnet.py]: yep, you can do it. See for example: http://www.martinos.org/mne/stable/generated/mne.Epochs.html#mne.Epochs

File: pyglmnet/pyglmnet.py (hunk 8 above)
Comment (jasmainak) [inline pyglmnet/pyglmnet.py]: I don't see it in the diff ... did you commit directly to master?

Comment (pavanramkumar) [inline pyglmnet/pyglmnet.py]: Yes, sorry. Once I merged the previous PR there were merge conflicts and I had to merge manually and push to master.

File: pyglmnet/pyglmnet.py (hunk 3 above)
Comment (pavanramkumar) [inline pyglmnet/pyglmnet.py]: This is awesome, thanks! Will keep in mind while doing a re-read of all docstrings
